src/sw.ts

```diff
@@ -476,6 +480,83 @@ async function fetchHandler ({ path, request, event }: FetchHandlerArg): Promise
   }
 }
 
+/**
+ * TODO: better styling
+ * TODO: more error details from @helia/verified-fetch
+ */
+async function errorPageResponse (fetchResponse: Response): Promise<Response> {
+  const responseContentType = fetchResponse.headers.get('Content-Type')
+  let json: Record<string, any> | null = null
+  let text: string | null = null
+  if (responseContentType != null) {
+    if (responseContentType.includes('text/html')) {
+      return fetchResponse
+    } else if (responseContentType.includes('application/json')) {
+      // we may eventually provide error messaging from @helia/verified-fetch
+      json = await fetchResponse.json()
+    } else {
+      text = await fetchResponse.text()
+      json = { error: { message: fetchResponse.statusText, stack: text } }
+    }
+  }
+  if (json == null) {
+    json = { error: { message: fetchResponse.statusText, stack: (new Error()).stack } }
+  }
+
+  /**
+   * TODO: output configuration
+   */
+  return new Response(`
+      <h1>Oops! Something went wrong inside of Service Worker IPFS Gateway.</h1>
+      <p><button onclick="window.location.reload(true);">Click here to retry</button></p>
+      <p>
+        <p>Error details:</p>
+        <p>${json.error.message}</p>
+        <pre>${json.error.stack}</pre>
+      </p>
+      <p>
+        <p>Response details:</p>
+        <pre>${JSON.stringify(json, null, 2)}</pre>
+      </p>
+      <p>
+        <p>Service worker details:</p>
+        <pre>${JSON.stringify(await getServiceWorkerDetails(), null, 2)}</pre>
+      </p>
+    `, {
+    status: fetchResponse.status,
+    statusText: fetchResponse.statusText,
+    headers: new Headers({
+      'Content-Type': 'text/html'
+    })
+  })
+}
+
+interface ServiceWorkerDetails {
+  config: ConfigDb
+  crossOriginIsolated: boolean
+  installTime: string
+  origin: string
+  scope: string
+  state: string
+}
+
+/**
+ * TODO: more service worker details
+ */
+async function getServiceWorkerDetails (): Promise<ServiceWorkerDetails> {
+  const registration = self.registration
+  const state = registration.installing?.state ?? registration.waiting?.state ?? registration.active?.state ?? 'unknown'
+
+  return {
+    config: await getConfig(),
+    crossOriginIsolated: self.crossOriginIsolated,
+    installTime: (new Date(firstInstallTime)).toISOString(),
```

Comment:
nit: UTC makes debugging easier
```suggestion
    installTime: (new Date(firstInstallTime)).toUTCString(),
```